packages/@aws-cdk/aws-codepipeline-actions/test/codestar-connections/codestar-connections-source-action.test.ts

```diff
@@ -148,27 +148,155 @@ describe('CodeStar Connections source Action', () => {
       ],
     });
 
+  });
+
+  test('exposes variables', () => {
+    const stack = new Stack();
+
+    createBitBucketAndCodeBuildPipeline(stack, {
+      triggerOnPush: false,
+    });
+
+    expect(stack).toHaveResourceLike('AWS::CodePipeline::Pipeline', {
+      'Stages': [
+        {
+          'Name': 'Source',
+        },
+        {
+          'Name': 'Build',
+          'Actions': [
+            {
+              'Name': 'CodeBuild',
+              'Configuration': {
+                'EnvironmentVariables': '[{"name":"CommitId","type":"PLAINTEXT","value":"#{Source_BitBucket_NS.CommitId}"}]',
+              },
+            },
+          ],
+        },
+      ],
+    });
 
   });
+
+  test('exposes variables with custom namespace', () => {
+    const stack = new Stack();
+
+    createBitBucketAndCodeBuildPipeline(stack, {
+      triggerOnPush: false,
+      variablesNamespace: 'kornicameister',
+    });
+
+    expect(stack).toHaveResourceLike('AWS::CodePipeline::Pipeline', {
+      'Stages': [
+        {
+          'Name': 'Source',
+          'Actions': [
+            {
+              'Name': 'BitBucket',
+              'Namespace': 'kornicameister',
+            },
+          ],
+        },
+        {
+          'Name': 'Build',
+          'Actions': [
+            {
+              'Name': 'CodeBuild',
+              'Configuration': {
+                'EnvironmentVariables': '[{"name":"CommitId","type":"PLAINTEXT","value":"#{kornicameister.CommitId}"}]',
+              },
+            },
+          ],
+        },
+      ],
+    });
+
+
+  });
+
+  test('fail if variable from unused action is referenced', () => {
+    const stack = new Stack();
+
+    createUnreferencedVariablePipeline(stack);
```

Comment:
I have to say, I don't love this new `createUnreferencedVariablePipeline()` helper function. From what I can tell, it's basically the same as `createBitBucketAndCodeBuildPipeline()`, just adds a reference to an unused action, which is kind of weird. It makes this test not read very well.

How about you change `createBitBucketAndCodeBuildPipeline()` to return the created CodePipeline, call it from this test, and then you add the unused source action and the build action referencing it directly in this test's body? I think that will make this test read better, and you'll be able to delete the `createUnreferencedVariablePipeline()` helper.

Reply:
Fixed, will land in next commit.